src/argilla/client/sdk/text_classification/api.py

```diff
@@ -58,6 +58,67 @@ def data(
         )
 
 
+def add_dataset_labeling_rule(
+    client: AuthenticatedClient,
+    name: str,
+    rule: LabelingRule,
+) -> Response[Union[LabelingRule, HTTPValidationError, ErrorMessage]]:
+    url = "{}/api/datasets/{name}/TextClassification/labeling/rules".format(
+        client.base_url, name=name
+    )
+
+    response = httpx.post(
+        url=url,
+        json={"query": rule.query, "label": rule.label},
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    return build_typed_response(response, LabelingRule)
+
+
+def update_dataset_labeling_rule(
+    client: AuthenticatedClient,
+    name: str,
+    rule: LabelingRule,
+) -> Response[Union[HTTPValidationError, ErrorMessage]]:
+    url = "{}/api/datasets/TextClassification/{name}/labeling/rules/".format(
+        client.base_url, name=name
+    )
+
+    response = httpx.patch(
+        url,
+        json={"query": rule.query, "label": rule.label},
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    return build_typed_response(response, rule.__class__)
+
+
+def delete_dataset_labeling_rule(
+    client: AuthenticatedClient,
+    name: str,
+    rule: LabelingRule,
+) -> Response[Union[LabelingRule, HTTPValidationError, ErrorMessage]]:
+    url = "{}/api/datasets/TextClassification/{name}/labeling/rules?{rule}".format(
```

Comment:
The way to build the path is the same that for the update method: passing the query in the path. The query is the rule key. Argilla does not support two rules with the same query.

Reply:
fixed